src/main/java/org/opentripplanner/graph_builder/module/osm/WayPropertySet.java

```diff
@@ -120,17 +112,29 @@ public WayProperties getDataForWay(OSMWithTags way) {
       }
     }
 
+    float forwardSpeed = getCarSpeedForWay(way, false);
+    float backSpeed = getCarSpeedForWay(way, true);
+    StreetTraversalPermission permission = rightResult.getPermission();
+
     WayProperties result = rightResult
       .mutate()
       .bicycleSafety(
-        rightResult.getBicycleSafetyFeatures().forward(),
-        leftResult.getBicycleSafetyFeatures().back()
+        rightResult.getBicycleSafetyFeatures() != null
+          ? rightResult.getBicycleSafetyFeatures().forward()
+          : defaultBicycleSafetyForPermission.apply(permission, forwardSpeed),
```

Comment:
```suggestion
        Optional.ofNullable(rightResult.getBicycleSafetyFeatures())
          .map(wp -> wp.forward())
          .orElseGet(defaultBicycleSafetyForPermission.apply(permission, forwardSpeed)),
```

I find this a bit more readable.

Perhaps you could move it into the `SafetyFeaures` record itself so you can reuse the code.

My point is that there should be less duplication.

Reply:
I feel like `SafetyFeatures` is perhaps not the right place for it because it's checking that if itself is null so you would have to give the instance of the `SafetyFeatures` as an argument and I don't know if it's appropriate to put too much logic inside a record. I did also think that the repetition here was a bit ugly but couldn't figure out a good place where to put it. I've changed this to use the `Optional` now.

Reply:
Sorry, Do not replace a language feature with a library call. Optional is meant for return statements - not to replace `if`s and conditional operators `test ? a : b` - please use the language...

Reply:
So do I revert the changes back to using the ternary operator?

Reply:
Yes

Reply:
Done in https://github.com/opentripplanner/OpenTripPlanner/pull/4484/commits/67a85499fe87fa9c23a76910a6585b388f453652